components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/webapp/self-registration-process.jsp

```diff
@@ -210,8 +211,22 @@
                 if (error != null) {
                     request.setAttribute("errorMsg", error.getDescription());
                     request.setAttribute("errorCode", error.getCode());
+                    if (passwordPatternErrorCode.equals(error.getCode())) {
+                        String i18Resource = IdentityManagementEndpointUtil.i18n(recoveryResourceBundle, error.getCode());
+                        if (!i18Resource.equals(error.getCode())) {
+                            request.setAttribute("errorMsg", i18Resource);
+                        }
+                        if (isSelfRegistrationWithVerification) {
+                            request.getRequestDispatcher("self-registration-with-verification.jsp").forward(request, response);
```

Comment:
"errorMsg", "errorCode", "self-registration-with-verification.jsp", "self-registration-without-verification.jsp"
Move string values to constants

Reply:
Fixed